### PR TITLE
Removing redundant code and fixes in reconstruction

### PR DESCRIPTION
**Pull request by mbroz84**
- Removing FDDDigitWriterSpec since its already there as DigitWriterSpec
- Fix in DigitReader  and ReconstructorSpec: was exiting after single event
- Adjusting AOD conversion to table definition in AnalysisDataModel: 4 charge amplitudes instead of 8

### CODE REVIEW

File: Detectors/FIT/FDD/workflow/src/DigitReaderSpec.cxx [hunk 1]
```diff
@@ -50,36 +61,24 @@ void DigitReader::run(ProcessingContext& pc)
...
-    TTree* digTree = (TTree*)digFile.Get(mDigitTreeName.c_str());
-    if (!digTree) {
-      LOG(FATAL) << "Failed to load FDD digits tree " << mDigitTreeName << " from " << mInputFileName;
+  mTree->SetBranchAddress(mDigitBCBranchName.c_str(), &digitsBC);
+  mTree->SetBranchAddress(mDigitChBranchName.c_str(), &digitsCh);
+
+  if (mUseMC) {
+    if (mTree->GetBranch(mTriggerBranchName.c_str())) {
+      mTree->SetBranchAddress(mTriggerBranchName.c_str(), &digitsTrig);
     }
```
Comment (shahor02) [inline Detectors/FIT/FDD/workflow/src/DigitReaderSpec.cxx]: why reading of the trigger input depends on requesting labels?

Comment (mbroz84) [reply Detectors/FIT/FDD/workflow/src/DigitReaderSpec.cxx]: Right, it should not be there, I moved it out

**Comment by shahor02** [comment]
@mbroz84 could you update dev and resolve conflicts?

**Comment by mbroz84** [comment]
> @mbroz84 could you update dev and resolve conflicts?

Done

**Comment by davidrohr** [comment]
@shahor02 : CI has passed now, can we merge it?

**Pull request merged**